tldw_Server_API/app/api/v1/schemas/archetype_schemas.py

```
@@ -0,0 +1,106 @@
+    Exactly one of ``template_key`` or ``custom`` must be provided.
+    """
+
+    template_key: str | None = None
+    custom: dict[str, Any] | None = None
+
+    @model_validator(mode="after")
+    def _exactly_one_source(self) -> ArchetypeStarterCommand:
+        has_template = self.template_key is not None
+        has_custom = self.custom is not None
+        if has_template == has_custom:
+            raise ValueError("Exactly one of 'template_key' or 'custom' must be provided")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)